File: olivemain/olive/consensus/block_rewards.py
from olive.util.ints import uint32, uint64

# 1 olive coin = 1,000,000,000,000 = 1 trillion dog.
_dog_per_olive = 1000000000000
_blocks_per_year = 1681920  # 32 * 6 * 24 * 365


def calculate_pool_reward(height: uint32) -> uint64:
    """
    Returns the pool reward at a certain block height. The pool earns 7/8 of the reward in each block. If the farmer
    is solo farming, they act as the pool, and therefore earn the entire block reward.
    These halving events will not be hit at the exact times
    (3 years, etc), due to fluctuations in difficulty. They will likely come early, if the network space and VDF
    rates increase continuously.
    """

    if height == 0:
        return uint64(int((7 / 8) * 21000000 * _dog_per_olive))
    elif height < (1 / 12) * _blocks_per_year:
        return uint64(int((7 / 8) * 100 * _dog_per_olive)) #Reward for early participants in the community.
    elif height < 2 * _blocks_per_year:
        return uint64(int((7 / 8) * 20 * _dog_per_olive))
    elif height < 4 * _blocks_per_year:
        return uint64(int((7 / 8) * 10 * _dog_per_olive))
    elif height < 6 * _blocks_per_year:
        return uint64(int((7 / 8) * 5 * _dog_per_olive))
    elif height < 8 * _blocks_per_year:
        return uint64(int((7 / 8) * 2.5 * _dog_per_olive))
    else:
        return uint64(int((7 / 8) * 1.25 * _dog_per_olive))


def calculate_base_farmer_reward(height: uint32) -> uint64:
    """
    Returns the base farmer reward at a certain block height.
    The base fee reward is 1/8 of total block reward

    Returns the coinbase reward at a certain block height. These halving events will not be hit at the exact times
    (3 years, etc), due to fluctuations in difficulty. They will likely come early, if the network space and VDF
    rates increase continuously.
    """
    if height == 0:
        return uint64(int((1 / 8) * 21000000 * _dog_per_olive))
    elif height < (1 / 12) * _blocks_per_year:
        return uint64(int((1 / 8) * 100 * _dog_per_olive)) #Reward for early participants in the community.
    elif height < 2 * _blocks_per_year:
        return uint64(int((1 / 8) * 20 * _dog_per_olive))
    elif height < 4 * _blocks_per_year:
        return uint64(int((1 / 8) * 10 * _dog_per_olive))
    elif height < 6 * _blocks_per_year:
        return uint64(int((1 / 8) * 5 * _dog_per_olive))
    elif height < 8 * _blocks_per_year:
        return uint64(int((1 / 8) * 2.5 * _dog_per_olive))
    else:
        return uint64(int((1 / 8) * 1.25 * _dog_per_olive))

File: olivemain/olive-blockchain-gui/src/util/blockRewards.ts
import Big from 'big.js';

const DOG_PER_olive = Big(1000000000000);
const BLOCKS_PER_YEAR = 1681920;

export function calculatePoolReward(height: number): Big {
  if (height === 0) {
    return DOG_PER_olive.times(21000000).times(7 / 8);
  }
  if (height < (1 / 12) * BLOCKS_PER_YEAR) {
    return DOG_PER_olive.times(100).times(7 / 8);
  }
  if (height < 2 * BLOCKS_PER_YEAR) {
    return DOG_PER_olive.times(20).times(7 / 8);
  }
  if (height < 4 * BLOCKS_PER_YEAR) {
    return DOG_PER_olive.times(10).times(7 / 8);
  }
  if (height < 6 * BLOCKS_PER_YEAR) {
    return DOG_PER_olive.times(5).times(7 / 8);
  }
  if (height < 8 * BLOCKS_PER_YEAR) {
    return DOG_PER_olive.times(2.5).times(7 / 8);
  }

  return DOG_PER_olive.times(1.25).times(7 / 8);
}

export function calculateBaseFarmerReward(height: number): Big {
  if (height === 0) {
    return DOG_PER_olive.times(21000000).times(1 / 8);
  }
  if (height < (1 / 12) * BLOCKS_PER_YEAR) {
    return DOG_PER_olive.times(100).times(1 / 8);
  }
  if (height < 2 * BLOCKS_PER_YEAR) {
    return DOG_PER_olive.times(20).times(1 / 8);
  }
  if (height < 4 * BLOCKS_PER_YEAR) {
    return DOG_PER_olive.times(10).times(1 / 8);
  }
  if (height < 6 * BLOCKS_PER_YEAR) {
    return DOG_PER_olive.times(5).times(1 / 8);
  }
  if (height < 8 * BLOCKS_PER_YEAR) {
    return DOG_PER_olive.times(2.5).times(1 / 8);
  }

  return DOG_PER_olive.times(1.25).times(1 / 8);
}
